### PR TITLE
ocrvs-4117 Advanced Search Bookmark Feature

### PR DESCRIPTION
**Pull request by sadmananik**
Farajaland PR - https://github.com/opencrvs/opencrvs-farajaland/pull/416

Peek 2022-11-24 01-17

### CODE REVIEW

File: packages/client/src/views/AdvancedSearch/RemoveBookmarkModal.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+}: IRemoveBookmarkModalProps) {
+  const intl = useIntl()
+  const dispatch = useDispatch()
+  const { searchId, ...rest } = useSelector(getPartialState)
+  const userDetails = useSelector(getUserDetails)
+
+  const [removeAdvancedSearchResultBookmark] = useMutation<
+    IRemoveBookmarkData,
+    MutationRemoveBookmarkedAdvancedSearchArgs
+  >(REMOVE_ADVANCED_SEARCH_RESULT_BOOKMARK_MUTATION, {
```
Comment (Zangetsu101) [inline packages/client/src/views/AdvancedSearch/RemoveBookmarkModal.tsx:59]: We should able to get both the types from `gateway.ts`

File: packages/client/src/views/AdvancedSearch/SaveBookmarkModal.tsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  const [queryName, setQueryName] = React.useState('')
+  const onChangeQueryName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value
+    setQueryName(value)
+  }
+
+  const [bookmarkAdvancedSearchResult] = useMutation<
+    ISaveBookmarkData,
+    MutationBookmarkAdvancedSearchArgs
+  >(BOOKMARK_ADVANCED_SEARCH_RESULT_MUTATION, {
```
Comment (Zangetsu101) [inline packages/client/src/views/AdvancedSearch/SaveBookmarkModal.tsx:78]: Same for this as well

File: packages/client/src/views/AdvancedSearch/RemoveBookmarkModal.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+              setNotificationMessages(
+                intl.formatMessage(
+                  messagesSearch.removeAdvancedSearchBookmarkLoadingNotification
+                )
+              )
+              setNotificationStatus(NOTIFICATION_STATUS.IN_PROGRESS)
+              toggleRemoveBookmarkModal()
+              await removeAdvancedSearchBookmarkHandler()
+            }}
+            disabled={false}
```
Comment (Zangetsu101) [inline packages/client/src/views/AdvancedSearch/RemoveBookmarkModal.tsx:129]: We should probably use the online status here instead

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/client/src/views/AdvancedSearch/BookmarkAdvancedSearchResult.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  }
+`
+export interface IBookmarkAdvancedSearch {
+  name: string
+  parameters: IAdvancedSearchParamState
+  userId: string
+}
+
+export function BookmarkAdvancedSearchResult() {
+  const advancedSearchState = useSelector(getPartialState)
```
[the remark this answers is not in the document]
Comment (sadmananik) [reply packages/client/src/views/AdvancedSearch/BookmarkAdvancedSearchResult.tsx:42]: Fixed

File: packages/client/src/views/AdvancedSearch/RemoveBookmarkModal.tsx (hunk 3 above)
Comment (sadmananik) [reply packages/client/src/views/AdvancedSearch/RemoveBookmarkModal.tsx:129]: Yeah

File: packages/client/src/components/interface/Navigation.tsx [hunk 5]
```diff
@@ -648,6 +661,37 @@ export const NavigationView = (props: IFullProps) => {
...
+                ) as IAdvancedSearchParamState
+                setAdvancedSearchParam({
+                  ...filteredParam,
+                  searchId: bookmarkResult.searchId
+                })
+                goToAdvancedSearchResultAction()
+              }}
+            />
+          )
+        })
```
[the remark this answers is not in the document]
Comment (sadmananik) [reply packages/client/src/components/interface/Navigation.tsx:691]: Yes.

**Review by Zangetsu101 — APPROVED** [review]
We are almost done with this. Just a couple of things:

- [x] ES_HOST needs to be added in the `docker-compose.deploy.yml` under migration service
- [ ] Can we use the type `AdvancedSearchParameters` (generated in gateway.ts) instead of `IAdvancedSearchParameters`?

**Pull request merged**